src/events/rewards/rewards.service.ts

```diff
@@ -135,7 +147,7 @@ export class RewardsService {
       fromBlock,
       address: res.address,
     });
-    const lastLog = logs[logs.length - 1];
+    const lastLog = logs[0];
```

Comment:
why do we take the first log from the array?

Reply:
that is because we search by blocks form `fromBlock` to `latest`, and last element will always the same.